pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -586,22 +586,21 @@ func (rc *reconciler) updateDevicePath(volumesNeedUpdate map[v1.UniqueVolumeName
 }
 
 func getDeviceMountPath(volume *reconstructedVolume) (string, error) {
-	volumeAttacher, err := volume.attachablePlugin.NewAttacher()
-	if volumeAttacher == nil || err != nil {
-		return "", err
-	}
-	deviceMountPath, err :=
-		volumeAttacher.GetDeviceMountPath(volume.volumeSpec)
-	if err != nil {
-		return "", err
-	}
-
+	var (
+		deviceMountPath string
+	)
 	if volume.blockVolumeMapper != nil {
-		deviceMountPath, err =
-			volume.blockVolumeMapper.GetGlobalMapPath(volume.volumeSpec)
-		if err != nil {
+		// for block volume, we return its global map path
+		return volume.blockVolumeMapper.GetGlobalMapPath(volume.volumeSpec)
+	} else if volume.attachablePlugin != nil {
+		// for filesystem volume, we return its device mount path if the plugin implements AttachableVolumePlugin
+		volumeAttacher, err := volume.attachablePlugin.NewAttacher()
+		if volumeAttacher == nil || err != nil {
 			return "", err
 		}
+		return volumeAttacher.GetDeviceMountPath(volume.volumeSpec)
+	} else {
+		return "", fmt.Errorf("blockVolumeMapper or attachablePlugin required")
 	}
 	return deviceMountPath, nil
```

Comment:
This should be really unreachable now.

Reply:
my mistake, `deviceMountPath` is removed